implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -104,41 +107,70 @@ impl NodeManager {
         } else {
             self.identity()?.async_try_clone().await?
         };
+        let provided_credential = if let Some(credential_name) = credential_name {
+            let encoded_cred = self
+                .cli_state
+                .credentials
+                .get(&credential_name)?
+                .config()
+                .await?
+                .encoded_credential;
+
+            let bytes = match hex::decode(encoded_cred) {
```

Comment:
A `get_credential_by_name` function on `credentials` returning a decoded credential directly would be nice.